src/pseudopeople/noise.py

```diff
@@ -51,6 +51,11 @@ def noise_dataset(
     randomness = get_randomness_stream(dataset.name, seed, dataset_data.index)
 
     noise_configuration = configuration[dataset.name]
+
+    # We only need to do this once, because noise does not introduce missingness,
+    # except for the leave_blank kind which is special-cased below
+    missingness = (dataset_data == "") | (dataset_data.isna())
```

Comment:
So the speedup is calculating missingness once for the entire dataframe instead of ad hoc as needed per column? It really sped things up that much?

Reply:
The old way was calculating it for each _noise type_ for each column. Even with the profiling I did it's a bit hard to directly measure the impact of this change since it changes the structure of things, but the time spent in the `isna` method was cut in half.